target/linux/ipq40xx/patches-5.4/202-arm-compressed-set-ipq40xx-watchdog-to-allow-boot.patch

```diff
@@ -0,0 +1,71 @@
+From 11d6a6128a5a07c429941afc202b6e62a19771be Mon Sep 17 00:00:00 2001
+From: John Thomson <git@johnthomson.fastmail.com.au>
+Date: Fri, 23 Oct 2020 19:42:36 +1000
+Subject: [PATCH 2/2] arm: compressed: set ipq40xx watchdog to allow boot
+
+For IPQ40XX systems where the SoC watchdog is activated before linux,
+the watchdog timer may be too small for linux to finish uncompress,
+boot, and watchdog management start.
+If the watchdog is enabled, set the timeout for it to 30 seconds.
+The functionality and offsets were copied from:
+drivers/watchdog/qcom-wdt.c qcom_wdt_set_timeout & qcom_wdt_start
+The watchdog memory address was taken from:
+arch/arm/boot/dts/qcom-ipq4019.dtsi
+
+This was required on Mikrotik IPQ40XX consumer hardware using Mikrotik's
+RouterBoot bootloader.
+
+Signed-off-by: John Thomson <git@johnthomson.fastmail.com.au>
+---
+ arch/arm/boot/compressed/head.S | 35 +++++++++++++++++++++++++++++++++
```

Comment:
Probably wants a different patch number to 200 to be in the right category?
Mybad.

Reply:
Yeah, need to see what category would appended_dtb and this one fit in actually.

Reply:
Could you move this to 302-?
Wants testing on watchdog not enabled at boot device (it should test that the watchdog is not enabled, and do nothing, but has not been tested), or a Kconfig conditional.

Reply:
Yeah, will change the numbering.
I can test on 2-3 more devices to make sure it does not break anything, I think that Qualcomm's U-boot also enables the WDT after bootipq command, but it sets 30 seconds.